cms/core/tests/test_settings.py

```diff
@@ -0,0 +1,67 @@
+from django.test import override_settings
+from wagtail.coreutils import get_dummy_request
+from wagtail.models import Site
+
+from cms.core.models.settings import SocialMediaSettings
+from cms.core.tests import TransactionTestCase
+
+
+class SocialSettingsTestCase(TransactionTestCase):
+    """Tests for the SocialSettings model, and how it behaves with multiple databases."""
+
+    def setUp(self):
+        self.request = get_dummy_request()
+
+        # Pre-warm site cache to avoid queries
+        self.site = Site.find_for_request(self.request)
+
+    def test_none_site(self):
+        """Test getting settings for a site which is None."""
+        with self.assertRaises(SocialMediaSettings.DoesNotExist):
+            SocialMediaSettings.for_site(None)
+
+    def test_create_setting(self):
+        """Test creating a setting."""
+        self.assertEqual(SocialMediaSettings.objects.count(), 0)
+
+        with self.assertNumQueriesConnection(default=4, replica=1):
+            SocialMediaSettings.for_request(self.request)
+
+        # Explicitly use default connection to work around strange issue with Django
```

Comment:
The model just doesn't appear in the DB. If I leave out `.using`, there's a count of 0, even though there's a clear DB query run. 

It might be another strange bug in Django, but being explicit here made it go away so I left it.